### PR TITLE
Implement very basic API searching

### LINKED ISSUE #1348: Add some search parameters when performing a GET on /incidents or /components
It would be great to be able to get incidents or components using for example, name or url parameters.
Currently, I need to parse all result (i.e. process all paginated results) and then search on it. 
What a waste of CPU ;)

### PR DESCRIPTION
**Pull request by jbrooksuk**
Closes #1348

---

@joecohens maybe something to re-visit so we can add things like between dates etc.

### CODE REVIEW

File: app/Http/Controllers/Api/ComponentController.php [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
 class ComponentController extends AbstractApiController
 {
+    use SearchableTrait;
```
Comment (joecohens) [inline app/Http/Controllers/Api/ComponentController.php]: Maybe we could do it on the model as we did with the sorting?

Comment (jbrooksuk) [inline app/Http/Controllers/Api/ComponentController.php]: Yeah, okay :)

**Comment by jbrooksuk** [comment]
@joecohens implemented on the model. Will merge once complete.

**Comment by joecohens** [comment]
AWESOME!!! :beers:

File: app/Models/Traits/SearchableTrait.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * @author James Brooks <james@alt-three.com>
+ */
+trait SearchableTrait
+{
+    /**
+     * Adds a sort scope.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param array                                 $column
```
Comment (GrahamCampbell) [inline app/Models/Traits/SearchableTrait.php]: oops

Comment (jbrooksuk) [inline app/Models/Traits/SearchableTrait.php]: Whoopsy. I'll fix.

**Pull request merged**